Library/src/Systems/Assets/AssetSystem.cpp

```diff
@@ -488,6 +487,63 @@ void AssetSystem::UpdateAssetCollectionMetadata(const AssetCollection& AssetColl
 	static_cast<chs::PrototypeApi*>(PrototypeAPI)->apiV1PrototypesIdPut(AssetCollection.Id, PrototypeInfo, ResponseHandler);
 }
 
+void AssetSystem::GetAssetCollectionCount(const csp::common::Optional<csp::common::Array<csp::common::String>>& Ids,
+										  const csp::common::Optional<csp::common::String>& ParentId,
+										  const csp::common::Optional<csp::common::Array<csp::common::String>>& Names,
+										  const csp::common::Optional<csp::common::Array<EAssetCollectionType>>& Types,
+										  const csp::common::Optional<csp::common::Array<csp::common::String>>& Tags,
+										  const csp::common::Optional<csp::common::Array<csp::common::String>>& SpaceIds,
+										  csp::systems::AssetCollectionCountResultCallback Callback)
+{
+	std::optional<std::vector<String>> PrototypeIds	  = Convert(Ids);
+	std::optional<String> ParentPrototypeId			  = Convert(ParentId);
+	std::optional<std::vector<String>> PrototypeNames = Convert(Names);
+
+
+	std::optional<std::vector<String>> PrototypeTypes;
+
+	if (Types.HasValue())
+	{
+		std::vector<String> Vals;
+
+		for (size_t i = 0; i < Types->Size(); ++i)
+		{
+			Vals.push_back(ConvertAssetCollectionTypeToString(Types->operator[](i)));
+		}
+
+		PrototypeTypes = std::move(Vals);
```

Comment:
[Consider] This whole block could be :
```suggestion
		std::transform(Types.value.cbegin(), Types.value.cend(), std::back_inserter(PrototypeTypes.value), [](const EAssetCollectionType& type) { return ConvertAssetCollectionTypeToString(type); });
```

Although ... bet this dosen't work because our interop types don't have properly conforming iterators. :(

Reply:
Yep :(

Reply:
We can update this with your recent change eventually

Reply:
Oh the iterator change, yeah would be nice. (Resolve yourself if you're not planning to address this now)